src/main/org/codehaus/groovy/transform/sc/StaticCompilationVisitor.java

```diff
@@ -123,9 +123,20 @@ public void visitClass(final ClassNode node) {
         }
         super.visitClass(node);
         addPrivateFieldAndMethodAccessors(node);
+        if (isStaticallyCompiled(node)) {
+            ClassNode outer = node.getOuterClass();
+            if (outer != null && !isStaticallyCompiled(outer)) {
+                outer.putNodeMetaData(StaticCompilationMetadataKeys.DYNAMIC_OUTER_NODE, true);
+            }
+        }
         classNode = oldCN;
     }
 
+    public void visitDynamicOuterClass(ClassNode node) {
```

Comment:
adding a public method here is not ideal, though it seems like the alternative would be to create a separate class responsible for adding bridge methods, which would be used by StaticCompilationVisitor as well as the phase operation added for dynamic outer classes.